koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -64,24 +64,31 @@ def __init__(
             raise AzureServiceError("Azure Service credentials are not configured.")
 
     def _get_latest_blob(
-        self, report_path: str, blobs: list[BlobProperties], extension: str
+        self, report_path: str, blobs: list[BlobProperties], extension: t.Optional[str] = None
     ) -> t.Optional[BlobProperties]:
+        valid_extensions = [
+            AzureBlobExtension.csv.value,
+            AzureBlobExtension.gzip.value,
+            AzureBlobExtension.json.value,
+        ]
         latest_blob = None
         for blob in blobs:
-            if not blob.name.endswith(extension):
+            if extension:
+                if not blob.name.endswith(extension):
+                    continue
+            elif not any(blob.name.endswith(ext) for ext in valid_extensions):
```

Comment:
Should we just drop this logic and the valid extensions list above? Essentially if we are not trying to download a manifest this `elif` statement is going to mangle file types together anyway so its kinda redundant no?

Reply:
Dropping this logic broke the tests, which indicates that this filtering is needed. Dropping this logic, we can lose control over the extensions used, I think. Maybe we can keep it to maintain the consistency and align with the existing tests...